### ISSUE TITLE
Spotty stops playing

### ISSUE THREAD

**Issue by sspeed22** [michaelherger/Spotty-Plugin#147]
I get often where Spotty stops playing.  v4.11.2 on LMS 8.5.3 (stable branch)  piCoreplayer 9.2.0 (rpi5)  Was getting the same error on a rpi4 with 8.2.0 and Lyrion on the development branch.

[24-11-01 08:44:04.5226] Slim::Player::StreamingController::playerEndOfStream (2324) 00:04:20:22:84:be
[24-11-01 08:44:04.5228] Slim::Player::StreamingController::_Invalid (324) 00:04:20:22:84:be: event EndOfStream received while in invalid state PLAYING-TRACKWAIT
[24-11-01 08:44:04.5229] Slim::Player::StreamingController::_Invalid (325) Error: 
[24-11-01 08:44:04.5232] Slim::Player::StreamingController::_Invalid (325) Backtrace:

   frame 0: Slim::Utils::Log::logBacktrace (/usr/local/slimserver/Slim/Player/StreamingController.pm line 325)
   frame 1: Slim::Player::StreamingController::_Invalid (/usr/local/slimserver/Slim/Player/StreamingController.pm line 285)
   frame 2: Slim::Player::StreamingController::_eventAction (/usr/local/slimserver/Slim/Player/StreamingController.pm line 2330)
   frame 3: Slim::Player::StreamingController::playerEndOfStream (/usr/local/slimserver/Slim/Player/Squeezebox2.pm line 150)
   frame 4: Slim::Player::Squeezebox2::statHandler (/usr/local/slimserver/Slim/Networking/Slimproto.pm line 676)
   frame 5: Slim::Networking::Slimproto::_disco_handler (/usr/local/slimserver/Slim/Networking/Slimproto.pm line 425)
   frame 6: Slim::Networking::Slimproto::client_readable (/usr/local/slimserver/Slim/Networking/IO/Select.pm line 122)
   frame 7: (eval) (/usr/local/slimserver/Slim/Networking/IO/Select.pm line 118)
   frame 8: Slim::Networking::IO::Select::__ANON__ (/usr/local/slimserver/Slim/Networking/IO/Select.pm line 167)
   frame 9: (eval) (/usr/local/slimserver/Slim/Networking/IO/Select.pm line 167)
   frame 10: Slim::Networking::IO::Select::loop (/usr/local/slimserver/slimserver.pl line 726)
   frame 11: main::idle (/usr/local/slimserver/slimserver.pl line 675)
   frame 12: main::main (/usr/local/slimserver/slimserver.pl line 1214)

**Comment by michaelherger** [comment]
I have a new build in development which seems to be much more stable: if you don't need the Connect feature, add http://herger.net/slim-plugins/spotty-test.xml to the additional repositories in Manage Plugins, then install the latest Spotty version from there.

**Comment by sspeed22** [comment]
I do use Connect, but I'll give the new version a try and go through the LMS interface instead.  If I leave the test repository in place   will a version with Connect get upgraded later?

**Comment by michaelherger** [comment]
Yes. LMS should always install the latest. Once I release an official update, it'll feature a more recent version number than the test version.

**Comment by alfredtorch** [comment]
Hi,

I don't know if you have the same issue as we have... sometimes the playback will loop at the end of a song. I created a script checking for this and for now, the music never stops (it skips to the next song).

Feel free to use it,
[songMonitorLMS.sh.txt](https://github.com/user-attachments/files/18395649/songMonitorLMS.sh.txt)

**Comment by FHFS** [comment]
I also have this probleml Playing the radio through LMS never stops.

Playing songs from spotify on LMS using Spotty will stop after a random amount of time. I can manually skip to the next song in the playlist, but having to do this every time is very inconvenient.


What logs can I collect to trap the cause of this issue?

**Comment by cregeland** [comment]
Same issue here - seems to happen more frequently on chromecast players using chromecast bridge.
